src/SELF_DGModel2D_t.f90

```diff
@@ -351,13 +348,13 @@ subroutine Write_DGModel2D_t(this,fileName)
       INFO("Writing control grid solution to file")
       call CreateGroup_HDF5(fileId,'/controlgrid')
       call this%solution%WriteHDF5(fileId,'/controlgrid/solution', &
-                                   this%decomp%offsetElem(this%decomp%rankId),this%decomp%nElem)
+                                   this%mesh%decomp%offsetElem(this%mesh%decomp%rankId),this%mesh%decomp%nElem)
 
       ! Write the geometry to file
       INFO("Writing control grid geometry to file")
       call CreateGroup_HDF5(fileId,'/controlgrid/geometry')
       call this%geometry%x%WriteHDF5(fileId,'/controlgrid/geometry/x', &
-                                     this%decomp%offsetElem(this%decomp%rankId),this%decomp%nElem)
+                                     this%mesh%decomp%offsetElem(this%mesh%decomp%rankId),this%mesh%decomp%nElem)
```

Comment:
These lines indicate that we need to test parallel writing of the 2D pickup files. I'll add an advection-diffusion_2d test that is run with mpi; that should get this covered.

Reply:
Resolved with [4f7a7a0](https://github.com/FluidNumerics/SELF/pull/53/commits/4f7a7a0e157b85dc677092583d328816442ade81)